integration/system/stats_test.go

```diff
@@ -161,32 +125,59 @@ func (s *DockerAPISuite) TestAPIStatsNetworkStats(c *testing.T) {
 		expRxPkts++
 		expTxPkts++
 	}
-	assert.Assert(c, postTxPackets >= expTxPkts, "Reported less TxPackets than expected. Expected >= %d. Found %d. %s", expTxPkts, postTxPackets, pingouts)
-	assert.Assert(c, postRxPackets >= expRxPkts, "Reported less RxPackets than expected. Expected >= %d. Found %d. %s", expRxPkts, postRxPackets, pingouts)
+	assert.Assert(t, postTxPackets >= expTxPkts, "Reported less TxPackets than expected. Expected >= %d. Found %d. %s", expTxPkts, postTxPackets, pingouts)
+	assert.Assert(t, postRxPackets >= expRxPkts, "Reported less RxPackets than expected. Expected >= %d. Found %d. %s", expRxPkts, postRxPackets, pingouts)
 }
 
-func (s *DockerAPISuite) TestAPIStatsNetworkStatsVersioning(c *testing.T) {
-	// Windows doesn't support API versions less than 1.25, so no point testing 1.17 .. 1.21
-	testRequires(c, testEnv.IsLocalDaemon, DaemonIsLinux)
+func TestAPIStatsNoStreamConnectedContainers(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
 
-	id := runSleepingContainer(c)
-	cli.WaitRun(c, id)
-	wg := sync.WaitGroup{}
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
 
-	for i := 17; i <= 21; i++ {
-		wg.Add(1)
-		go func(i int) {
-			defer wg.Done()
-			apiVersion := fmt.Sprintf("v1.%d", i)
-			statsJSONBlob := getVersionedStats(c, id, apiVersion)
-			if versions.LessThan(apiVersion, "v1.21") {
-				assert.Assert(c, jsonBlobHasLTv121NetworkStats(statsJSONBlob), "Stats JSON blob from API %s %#v does not look like a <v1.21 API stats structure", apiVersion, statsJSONBlob)
-			} else {
-				assert.Assert(c, jsonBlobHasGTE121NetworkStats(statsJSONBlob), "Stats JSON blob from API %s %#v does not look like a >=v1.21 API stats structure", apiVersion, statsJSONBlob)
-			}
-		}(i)
+	cID1 := container.Run(ctx, t, apiClient, container.WithImage("busybox"), container.WithCmd("sleep", "20"))
+	cID2 := container.Run(ctx, t, apiClient, container.WithImage("busybox"), container.WithCmd("top"), container.WithNetworkMode("container:"+cID1))
+
+	ch := make(chan error, 1)
+	go func() {
+		resp, body, err := request.Get(testutil.GetContext(t), "/containers/"+cID2+"/stats?stream=false")
+		defer body.Close()
+		if err != nil {
+			ch <- err
+		}
+		if resp.StatusCode != http.StatusOK {
+			ch <- fmt.Errorf("Invalid StatusCode %v", resp.StatusCode)
+		}
+		if resp.Header.Get("Content-Type") != "application/json" {
+			ch <- fmt.Errorf("Invalid 'Content-Type' %v", resp.Header.Get("Content-Type"))
+		}
+		var v *types.Stats
+		if err := json.NewDecoder(body).Decode(&v); err != nil {
+			ch <- err
+		}
+		ch <- nil
+	}()
+
+	select {
+	case err := <-ch:
+		assert.NilError(t, err, "Error in stats Engine API: %v", err)
+	case <-time.After(15 * time.Second):
+		t.Fatalf("Stats did not return after timeout")
 	}
-	wg.Wait()
+}
+
+func TestAPIStatsContainerNotFound(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
```

Comment:
Also curious why this one wouldn't work on Windows (perhaps it does?)

Reply:
[No idea](https://github.com/moby/moby/commit/66be81b148a1c6693c44cf52d8172d512eedc233#diff-6c2ce38ed3ae3f3b692546e8a9f06223a1b7dbd8e4d5a700f622eee7b3a8d2f7R139)! I don't have a Windows machine around, but maybe I'll get CI to check if it works.